tree/ntupleutil/v7/src/RNTupleImporterCLI.cxx

```diff
@@ -0,0 +1,197 @@
+   "    If not specified, the default size of 512 * 1024 * 1024 bytes is used.\n"
+   "  --convert-dots\n"
+   "    Whether to convert dots in branch names (if present) to underscores in field names.\n"
+   "    RNTuple does not allow for dots in field names, so this option will convert them to underscores.\n"
+   "    If not specified, no conversion happens and an error is thrown when branch names with dots are encountered.\n"
+   "  --verbose, -v\n"
+   "    Whether to print schema information and progress.\n"
+   "    If not specified, nothing will be printed except for a brief report about the source TTree and target "
+   "RNTuple.\n";
+
+ImporterConfig ROOT::Experimental::RNTupleImporterCLI::ParseArgs(const std::vector<std::string> &args)
+{
+   // Print help message and exit if "--help"
+   const auto argsProvided = args.size() >= 2;
+   const auto helpUsed = argsProvided && (args[1] == "--help" || args[1] == "-h");
+
+   if (!argsProvided || helpUsed) {
+      std::cout << usageText;
+      if (!argsProvided)
+         std::cout << " Use --help or -h for usage help.";
+      if (helpUsed)
+         std::cout << argUsageText;
+      std::cout << std::endl;
+
+      return {};
+   }
+
+   enum class EArgState {
+      kNone,
+      kTreeName,
+      kTreePath,
+      kNTuplePath,
+      kNTupleName,
+      kCompression,
+      kUnzippedPageSize,
+      kUnzippedClusterSize,
+      kZippedClusterSize
+   } argState = EArgState::kNone;
+   ImporterConfig config;
+   // The config will overwrite the write options set by `RNTupleImporter`, so we need to set the default zstd
+   // compression here as well.
+   config.fNTupleOpts.SetCompression(config.kDefaultCompressionSettings);
+
+   for (size_t i = 1; i < args.size(); ++i) {
+      const auto &arg = args[i];
+
+      if (arg == "--ttree" || arg == "-t") {
+         argState = EArgState::kTreeName;
+      } else if (arg == "--infile" || arg == "-i") {
+         argState = EArgState::kTreePath;
+      } else if (arg == "--outfile" || arg == "-o") {
+         argState = EArgState::kNTuplePath;
+      } else if (arg == "--rntuple" || arg == "-r") {
+         argState = EArgState::kNTupleName;
+      } else if (arg == "--compression" || arg == "-c") {
+         argState = EArgState::kCompression;
+      } else if (arg == "--unzipped-page-size") {
+         argState = EArgState::kUnzippedPageSize;
+      } else if (arg == "--zipped-cluster-size") {
+         argState = EArgState::kZippedClusterSize;
+      } else if (arg == "--max-unzipped-cluster-size") {
+         argState = EArgState::kUnzippedClusterSize;
+      } else if (arg == "--convert-dots") {
+         config.fConvertDots = true;
+      } else if (arg == "--verbose" || arg == "-v") {
+         config.fVerbose = true;
+      } else if (arg[0] == '-') {
+         std::cerr << "Unrecognized option '" << arg << "'\n";
+         return {};
+      } else {
```

Comment:
I think there is a subtle issue here: what if one of the arguments is named as an option, e.g. if we have a ROOT file names "-v" (I know... it's artificial, but still).

Not for this PR, but I think with options parsing implemented in several places (e.g. readspeed, but also others), I think we should have a common options parsing class à la `optarg`.